features/frameworks/greentea-client/source/test_env.cpp

```diff
@@ -61,8 +61,10 @@ static void greentea_notify_version();
  * Rawserial object used to provide direct, raw serial communications
  * between the target and the host.
  */
-RawSerial greentea_serial(USBTX, USBRX); 
-
+RawSerial& greentea_serial() { 
+    static RawSerial serial(USBTX, USBRX);
```

Comment:
This would be very good to introduce.

Perhaps we need a new pr with the thread-safety introduced + tests. Maybe deprecated the `SingletonPtr` for the language-builtin solution.

We may want to wait for the thread-safety first, just to avoid introducing temporary race conditions (unless we're sure these aren't used from multiple threads).